### DISCUSSION TITLE
Enhancement to the Profiles functionality.

### DISCUSSION THREAD

**Discussion by keywok** [runelite/runelite#17346] [Ideas]
I had a search through both the Issues and Discussions and couldn't see another post about this but is something that I know especially a lot of skillers would like to see.

The main thing stopping a lot of us using profiles is the fact they don't remembering where UI elements have been moved too. For example if you have a profile with Runecraft settings and have the inventory moved to be closer to the bank interface and another profile where the inventory is in the default position swapping between the two profiles won't move the inventory back and forth.

From my understanding and brief discussion within the RuneLite discord this is something that is stored within a profile however is only loaded upon the client being loaded for the first time and not when selecting a profile within the side panel. Having to reload a client to have the UI elements move is just slower than doing it manually and IMO defeats the purpose of having the ability to swap plugin settings in a single click.

Hopefully this makes sense but if not I can provide more information to illustrate the request further.

**Comment by Felanbird** [comment]
This already works, it just doesn't work when your inventory isn't touching one of the anchor points.

![RuneLite_2024-01-07_14-41-46](https://github.com/runelite/runelite/assets/41973452/1e7105b2-9f65-49c5-b05c-933e3b803a16)
In this I have the inventory attached to the anchor point above the chatbox, and the one left of the inventory's default position.
When the inventory in the `New Profile` profile is touching the anchor, it correctly swaps when switching, when it's reset to the default location `alt+right-click` it does not swap.

![RuneLite_2024-01-07_14-48-37](https://github.com/runelite/runelite/assets/41973452/4de90aed-b700-4166-bd6e-5b6d3d8eeb83)
For this one I just used the same anchor, but for `New Profile` I put the bottom right corner of the inventory on the top left corner of the anchor, and the one for `actual default` is just touching the anchor with the left side of the inventory.

**Reply by keywok** [reply]
Thats good to know but sadly the position that is mostly used for the example I gave, Runecrafting, isn't anywhere near an anchor point making the above not solve the problem. I have attached an image as an example of the position that is used when Runecrafting as it allows for much quicker banking with filling the pouch with essence.

![image](https://github.com/runelite/runelite/assets/71525889/ebbc3bc7-955b-479f-ba7c-31c9c08de88a)

**Reply by Felanbird** [reply]
Is that true? Isn't there an anchor directly under the inventory in that screenshot?

**Reply by keywok** [reply]
The only two anchors near would either cover the deposit bank interface which is needed and also put the pouch way too high or position the inventory in the middle of the client making it very out of place for depositing and also crafting runes on the altar.

![RuneLite_07-01-24_20-28-10](https://github.com/runelite/runelite/assets/71525889/69872361-0122-4a5e-acd3-cba1d36d0be3)

**Reply by Felanbird** [reply]
At the start of the gif I can see that your inventory is ~~_touching_~~ on-top of the anchor point, which is all that was required to do in my testing shown above. Try switching to a separate profile that does not have the inventory in the default spot. 

e.g. like my ugly paint crop
![lazy-edit](https://github.com/runelite/runelite/assets/41973452/0d3f3639-3e35-483c-9ba4-eed2b39e7295)


edit: it's possible my overlays are slightly broken but in a good way, since mine don't snap (which happens to work out for your exact scenario) but I'm unsure why

**Reply by keywok** [reply]
Even with the moved inventory on-top of an anchor point but not attached to it after a single switch between the profile with it moved above an anchor point and a profile with it in the default position the inventory stops being moved.

The two profiles are identical in every way apart from the positioning of the inventory being default or as you suggested "touching or on-top" of an acnhor point.

https://github.com/runelite/runelite/assets/71525889/ecfb792a-9650-4b54-98a2-a1d78dd55bb3

**Reply by Felanbird** [reply]
In this video you are using `alt+right-click` to force the inventory back to the default position, that does _not_ work. 
What I wanted you to try was moving it manually to the spot I indicated in the image on a second profile (which is slightly away from the default position),
but that may not be possible as it seems I have an issue with snapping, but that issue happens to work out perfectly for the outcome _you_ want.

**Reply by keywok** [reply]
So it appears if you use the "Restore to default position" shortcut as you described it doesn't work but if you manually move it even to where the default position is it does. For this use case yes it does now work as I was hoping however feels very work-around esc rather than the functionality working as intended if that makes sense.

**Reply by Felanbird** [reply]
Yes I do agree it seems like a problem that without anchoring the position is not saved, I'm just not sure how that would be handled.

**Reply by keywok** [reply]
Even with the inventory where it is by default moved there manually without using the reset it works perfectly so it's most likely something to do with how resetting the element is handled in regards to the profiles.

**Reply by Felanbird** [reply]
Ah yes I can replicate that, I was previously doing all my testing with the inventory on one of the profiles being reset, which is why I was able to confirm that didn't work. What you've reported with 2 different manually-set location's works for me as well. (one of them being set on the very spot the default location is)

**Reply by keywok** [reply]
Well I'm glad we figured out a work-around and identified the cause of the bug.

**Comment by Adam-** [comment]
this has been marked as fixed by #17354
